ui/mage/fire/sim.ts

```diff
@@ -162,6 +169,54 @@ export class FireMageSimUI extends IndividualSimUI<Spec.SpecFireMage> {
 			new ReforgeOptimizer(this, {
 				experimental: true,
 				statSelectionPresets: Presets.FIRE_BREAKPOINTS,
+				updateSoftCaps: softCaps => {
+					const hasBL = !!player.getRaid()?.getBuffs()?.bloodlust;
+					const hasPI = !!player.getBuffs().powerInfusionCount;
+					const hasBerserking = player.getRace() === Race.RaceTroll;
+
+					const modifyHaste = (rating: number, modifier: number) =>
+						Math.round(
```

Comment:
No, I'm not sure now 😅
However when calculating breakpoints I always need to `round` and not `ceil` the value to get the correct value. In theory it shouldn't matter, but it could set the breakpoint +1 where it doesn't need to.